tests/cases/string/ltrim.test

```diff
@@ -10,7 +10,28 @@ ltrim(''::str, ' '::str) = ''::str
 ltrim(' '::str, ' '::str) = ''::str
 ltrim(null::str, ' '::str) = null::str
 
+# spaces_only: Examples with only spaces to trim off
+ltrim('abc'::str) = 'abc'::str
+ltrim('   abc'::str) = 'abc'::str
+ltrim('abc   '::str) = 'abc   '::str
+ltrim('   abc   '::str) = 'abc   '::str
+ltrim(''::str) = ''::str
+ltrim(' '::str) = ''::str
+ltrim(null::str) = null::str
```

Comment:
Should we test with some other potential defaults to make sure that folks aren't also trimming tabs and/or other whitespace as a default?  (and below and two other files)